gitgud/skills/user_messages.py

```diff
@@ -177,3 +177,74 @@ def display_entry(index, human_name, code_name, indent):
                 code_name=item.name,
                 indent=indent
             )
+
+
+def target_branch_str():
+    file_operator = operations.get_operator()
+    tree = file_operator.get_current_tree()
+    referred_by = {}
+    for branch_name in tree['branches']:
+        target = tree['branches'][branch_name]['target']
+        if target not in referred_by:
+            referred_by.update({target: [branch_name]})
+        else:
+            referred_by[target].append(branch_name)
+    for target in referred_by:
+        referred_by[target] = ", ".join(referred_by[target])
+    return referred_by
+
+
+def display_tree_data(header, data, show_content, file_count=2):
+    file_format_str = "  {path} - {content}"
+
+    def existence_str(condition):
+        if condition:
+            return "Exists"
+        else:
+            return "Doesn't exist"
+
+    data_paths = list(data.keys())
+    # If there are more files than we requested, we want to show that too
+
+    print(header + ":")
+    for index in range(max(file_count, len(data_paths))):
+        file_exists = index < len(data_paths)
+        # Handle number of tracked files
+        if file_exists:
+            path = data_paths[index]
+        else:
+            path = "File " + str(index + 1)
+        if show_content and file_exists:
+            content = data[data_paths[index]]
+        else:
+            content = existence_str(file_exists)
+        print(file_format_str.format(path=path, content=content))
+
+
+def display_commit_content(show_branches=True, show_content=True, file_count=2):  # noqa: E501
+    file_operator = operations.get_operator()
+    referred_by = target_branch_str()
+
+    commit_format_str = "{message}"
+    if show_branches:
+        commit_format_str += " {branches}"
+    commit_format_str += ":"
+
+    displayed = set()
+    for head in file_operator.repo.heads:
+        for commit in file_operator.repo.iter_commits(head, reverse=True):
+            if commit not in displayed:
+                if commit.hexsha in referred_by and show_branches:
```

Comment:
I'd reverse these conditions because show_branches is more important.

`show_branches and commit.hexsha in referred_by`